packages/enzyme-adapter-utils/src/Utils.js

```diff
@@ -92,24 +92,55 @@ export function nodeTypeFromType(type) {
   return 'function';
 }
 
-function isIterable(obj) {
-  return (
-    obj != null &&
-    typeof Symbol === 'function' &&
-    typeof Symbol.iterator === 'symbol' &&
-    typeof obj[Symbol.iterator] === 'function'
+function getIteratorFn(obj) {
+  const iteratorFn = obj && (
+    (
+      typeof Symbol === 'function' &&
+      typeof Symbol.iterator === 'symbol' &&
+      obj[Symbol.iterator]
+    ) ||
+    obj['@@iterator']
   );
+
+  if (typeof iteratorFn === 'function') {
+    return iteratorFn;
+  }
+
+  return undefined;
+}
+
+function isIterable(obj) {
+  return Boolean(getIteratorFn(obj));
 }
 
 export function isArrayLike(obj) {
   return Array.isArray(obj) || (isIterable(obj) && typeof obj !== 'string');
```

Comment:
(predates your PR, but still) let's flip the typeof check here; `isIterable` shouldn't be called at all in the string case.